api/src/nlp/controllers/nlp-sample.controller.ts

```diff
@@ -369,33 +371,7 @@ export class NlpSampleController extends BaseController<
     return deleteResult;
   }
 
-  /**
-   * Imports NLP samples from a CSV file.
-   *
-   * @param file - The file path or ID of the CSV file to import.
-   *
-   * @returns A success message after the import process is completed.
-   */
-  @CsrfCheck(true)
-  @Post('import/:file')
-  async import(
-    @Param('file')
-    file: string,
-  ) {
-    // Check if file is present
-    const importedFile = await this.attachmentService.findOne(file);
-    if (!importedFile) {
-      throw new NotFoundException('Missing file!');
-    }
-    const filePath = importedFile
-      ? join(config.parameters.uploadDir, importedFile.location)
-      : undefined;
-
-    // Check if file location is present
-    if (!fs.existsSync(filePath)) {
-      throw new NotFoundException('File does not exist');
-    }
-
+  private async parseAndSaveDataset(data: string) {
```

Comment:
I suggest we move this method to the service. Let's add a documentation + unit tests